front/src/modules/trainschedule/components/ManageTrainSchedule/Allowances/AllowancesActions.tsx

```diff
@@ -74,10 +72,15 @@ export default function AllowancesActions({
   const [endPosition, setEndPosition] = useState(pathLength);
   const [allowanceLength, setAllowanceLength] = useState(endPosition - beginPosition);
   const [distribution, setDistribution] = useState(distributionsList[0].value);
-  const [valueAndUnit, setValueAndUnit] = useState<AllowanceValue>();
+  const [valueAndUnit, setValueAndUnit] = useState<AllowanceValueForm>();
   const [isValid, setIsValid] = useState(false);
 
-  function validityTest() {
+  const allowanceValue = useMemo(() => {
+    if (valueAndUnit) return getAllowanceValue(valueAndUnit);
+    return undefined;
```

Comment:
the function is considered to return undefined if there is no return statement

Reply:
I have this error if I remove the return undefined, so I will keep it like this if it's ok for you
![image](https://github.com/osrd-project/osrd/assets/45000526/38648201-3b72-467d-a54c-5b4b22be4f1a)

Reply:
yes